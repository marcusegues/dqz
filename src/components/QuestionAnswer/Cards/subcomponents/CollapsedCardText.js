// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
import { size } from '../../../../styles/fonts';
import { CARD_GREY } from '../../../../styles/colors';

type CollapsedCardTextProps = {
  text: string,
};

export const CollapsedCardText = ({ text }: CollapsedCardTextProps) => (
  <Text
    style={{
      fontSize: size.small,
      color: CARD_GREY,
    }}
  >
    {text}
  </Text>
);
