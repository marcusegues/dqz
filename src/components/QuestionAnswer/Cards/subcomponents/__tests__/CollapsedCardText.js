import React from 'react';
import renderer from 'react-test-renderer';
import { CollapsedCardText } from '../CollapsedCardText';

describe('CollapsedCardText', () => {
  test('CollapsedCardText renders according to snapshot', () => {
    const component = renderer
      .create(<CollapsedCardText text="MockCollapsedCardText" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test('CollapsedCardText component renders correctly if color prop is set', () => {
    const tree = renderer
      .create(
        <CollapsedCardText
          text="MockCollapsedCardText"
          style={{ color: 'red' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
