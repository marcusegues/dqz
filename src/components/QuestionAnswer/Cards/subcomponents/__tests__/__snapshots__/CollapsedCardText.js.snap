// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsedCardText CollapsedCardText component renders correctly if color prop is set 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "red",
      "fontSize": 12,
    }
  }
>
  MockCollapsedCardText
</Text>
`;

exports[`CollapsedCardText CollapsedCardText renders according to snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#4A4A4A",
      "fontSize": 12,
    }
  }
>
  MockCollapsedCardText
</Text>
`;
